src/Utilities/Compiler/Extensions/IDictionaryExtensions.cs

```diff
@@ -28,5 +29,9 @@ internal static class IDictionaryExtensions
                 dictionary.Add(item);
             }
         }
+
+        public static bool IsEqualTo<TKey, TValue>(this IReadOnlyDictionary<TKey, TValue> dictionary, IReadOnlyDictionary<TKey, TValue> other)
```

Comment:
I have seen that for `ImmutableHashSet` the method was called `SetEquals` to I was wondering whether to call this `DictionaryEquals`. Let me know your preference.